tests/integration/config/plugins/integration-tests/data/tests/tests_mission_ships.txt

```diff
@@ -0,0 +1,221 @@
+test-data "Checking For A Gifted Ship Test Save"
+	category "savegame"
+	contents
+		pilot Bobbi Bughunter
+		date 16 11 3013
+		system Sol
+		planet Earth
+		ship "Star Barge"
+			name "Buggy Barge"
+			sprite "ship/star barge"
+			attributes
+				category "Light Freighter"
+				cost 190000
+				mass 70
+				bunks 3
+				"cargo space" 50
+				drag 2.1
+				"engine capacity" 40
+				"fuel capacity" 300
+				"heat dissipation" 0.8
+				hull 1000
+				"outfit space" 130
+				"required crew" 1
+				shields 600
+				"turret mounts" 1
+				"weapon capacity" 20
+			outfits
+				"X1700 Ion Thruster"
+				"X1200 Ion Steering"
+				Hyperdrive
+				"Anti-Missile Turret"
+				"D14-RN Shield Generator"
+				"nGVF-BB Fuel Cell"
+				"LP036a Battery Pack"
+			crew 1
+			fuel 300
+			shields 600
+			hull 1000
+			engine -9 38 1
+			engine 9 38 1
+			turret 0 -8 "Anti-Missile Turret"
+			leak leak 60 50
+			explode "tiny explosion" 10
+			explode "small explosion" 10
+			system Sol
+			planet Earth
+		account
+			credits 0
+			score 400
+			history
+
+test-data "Checking For A Gifted Ship Test Missions"
+	category mission
+	contents
+		mission "%TEST%: Switch"
+			landing
+			on offer
+				take ship "Star Barge"
+				give ship "Sparrow" "The Sparrow"
+					id "ship: sparrow"
+		mission "%TEST%: Check sparrow"
+			landing
+			on offer
+				take ship "Sparrow" "The Sparrow"
+					id "ship: sparrow"
+
+test "Checking For A Gifted Ship"
+	status active
+	description "Test losing a freighter, getting a sparrow and then verify the player has said (named) sparrow, after reloading the save."
+		# Create/inject the savegame and load it.
+		inject "Checking For A Gifted Ship Test Missions"
+		inject "Checking For A Gifted Ship Test Save"
+		call "Load First Savegame"
+		# Make the missions trigger by landing again.
+		call "Depart"
+		call "Land"
+		call "Depart"
+		# Reload this to make sure the UUID is well stored into the savefile.
+		call "Open Menu"
+		call "Load First Savegame"
+		call "Depart"
+		call "Land"
+		assert
```

Comment:
Make a few more assertions at the earlier steps?

Reply:
True, I'll make sure the player did get the ship